tests/e2e/core-tests/specs/api/variable-product.test.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable jest/no-export, jest/no-disabled-tests */
+/**
+ * Internal dependencies
+ */
+const { HTTPClientFactory, VariableProduct, ProductVariation } = require( '@woocommerce/api' );
+
+/**
+ * External dependencies
+ */
+const config = require( 'config' );
+const {
+	it,
+	describe,
+	beforeAll,
+} = require( '@jest/globals' );
+
+/**
+ * Create a variable product and retrieve via the API.
+ */
+const runVariableProductAPITest = () => {
+	describe('REST API > Variable Product', () => {
+		let client;
+		let defaultVariableProduct;
+		let defaultVariations;
+		let baseVariableProduct;
+		let product;
+		let variations = [];
+		let productRepository;
+		let variationRepository;
+
+		beforeAll(async () => {
+			defaultVariableProduct = config.get('products.variable');
+			defaultVariations = config.get('products.variations');
+			const admin = config.get('users.admin');
+			const url = config.get('url');
+
+			client = HTTPClientFactory.build(url)
+				.withBasicAuth(admin.username, admin.password)
+				.withIndexPermalinks()
+				.create();
+		});
+
+		it('can create a variable product', async () => {
+			productRepository = VariableProduct.restRepository(client);
+
+			// Check properties of product in the create product response.
+			product = await productRepository.create(defaultVariableProduct);
+			expect(product).toEqual(expect.objectContaining(defaultVariableProduct));
+		});
+
+		it('can add variations', async () => {
```

Comment:
This test doesn't run any asserts on the created variations, is there any we wanted to add here?

Reply:
Added an assertion for each variation.

Reply:
Thanks!